includes/class-sensei-quiz.php

```diff
@@ -1523,6 +1528,53 @@ public static function start_quiz_questions_loop() {
 		$sensei_question_loop['quiz_id']   = $quiz_id;
 	}
 
+	/**
+	 * Display quiz progress bar.
+	 *
+	 * @param int   $quiz_id quiz id.
+	 * @param array $pagination_settings pagination settings.
+	 * @since 3.15.0
+	 */
+	public static function display_progress_bar( $quiz_id, $pagination_settings ) {
+
+		$user_id             = get_current_user_id();
+		$lesson_id           = Sensei()->quiz->get_lesson_id( $quiz_id );
+		$all_questions_count = count( Sensei()->lesson->lesson_quiz_questions( $quiz_id, 'publish' ) );
+		$answers             = Sensei()->quiz->get_user_answers( $lesson_id, $user_id );
+		$answers_count       = is_array( $answers ) ? count(
+			array_filter(
+				$answers,
+				function( $item ) {
+					return is_array( $item );
+				}
+			)
+		) : 0;
+		Sensei()->assets->register( 'sensei-quiz-progress', 'blocks/progress-bar.js' );
```

Comment:
There is a setting that disables the JS in the front-end. To account for that, you can move the `Sensei()->assets->register( 'sensei-quiz-progress', 'blocks/progress-bar.js'` to `Sensei_Frontend::enqueue_scripts`.

Reply:
https://github.com/Automattic/sensei/pull/4606/files#diff-0ace0c95702bd1dbe3643c412535679aa487534e1d61852ceb1bbf43fc14de23R146